launch/test/launch/actions/test_log_info.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2020 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for the LogInfo action class."""
+
+import io
+import textwrap
+
+from launch import LaunchContext
+from launch.actions import LogInfo
+from launch.frontend import Parser
+from launch.utilities import perform_substitutions
+
+
+def test_log_info_constructors():
+    """Test the constructors for LogInfo class."""
+    LogInfo(msg='')
+    LogInfo(msg='foo')
+    LogInfo(msg=['foo', 'bar', 'baz'])
+
+
+def test_log_info_methods():
+    """Test the methods of the LogInfo class."""
+    launch_context = LaunchContext()
+
+    log_info = LogInfo(msg='')
+    assert perform_substitutions(launch_context, log_info.msg) == ''
+
+    log_info = LogInfo(msg='foo')
+    assert perform_substitutions(launch_context, log_info.msg) == 'foo'
+
+    log_info = LogInfo(msg=['foo', 'bar', 'baz'])
+    assert perform_substitutions(launch_context, log_info.msg) == 'foobarbaz'
+
+
+def test_log_info_execute():
+    """Test the execute (or visit) of the LogInfo class."""
+    log_info = LogInfo(msg='foo')
+    launch_context = LaunchContext()
+    assert log_info.visit(launch_context) is None
+
+
+def test_frontend():
+    launch_context = LaunchContext()
+    xml_file = \
+        """\
+        <launch>
+            <log message="Hello world!" />
+        </launch>
+        """
+    xml_file = textwrap.dedent(xml_file)
+    root_entity, parser = Parser.load(io.StringIO(xml_file))
+    launch_description = parser.parse_description(root_entity)
+    log_info = launch_description.entities[0]
+    assert isinstance(log_info, LogInfo)
+    assert perform_substitutions(launch_context, log_info.msg) == 'Hello world!'
```

Comment:
@jacobperron tests for `launch` actions XML descriptions are all found in `launch_xml`, not here. 

I'll go ahead an guess the Rpr job is failing because of entrypoint shenanigans. Consider moving this test to `launch_xml` and (now that we focus here) add an equivalent one to `launch_yaml` :smiley:

Reply:
Ah, good call. I'll try moving the tests :upside_down_face: 
I guess it worked locally because `launch_xml` was in my Rolling underlay.

Reply:
Fixed in 19ee578